src/state.rs

```diff
@@ -2,8 +2,26 @@
 
 use amethyst_input::is_close_requested;
 use ecs::prelude::World;
+use std::fmt::Result as FmtResult;
+use std::fmt::{Display, Formatter};
 use {GameData, StateEvent};
 
+#[derive(Debug)]
+pub enum StateError {
```

Comment:
Missing documentation explaining what it is. (its obvious, but we prefer to have docs as much as possible)